Include/internal/pycore_interp_structs.h

```diff
@@ -566,6 +566,8 @@ struct types_state {
         managed_static_type_state initialized[_Py_MAX_MANAGED_STATIC_EXT_TYPES];
     } for_extensions;
     PyMutex mutex;
+    // used to check correct usage of the above mutex
+    unsigned long long mutex_tid;
```

Comment:
This is used only in free-threading builds right? I think we should guard it with `ifdef Py_GIL_DISABLED`.

Reply:
Done.